lib/Server.js

```diff
@@ -644,14 +643,12 @@ class Server {
       // - https://github.com/nodejs/node/issues/21665
       // - https://github.com/webpack/webpack-dev-server/issues/1449
       // - https://github.com/expressjs/express/issues/3388
-      if (semver.gte(process.version, '10.0.0') || !options.spdy) {
+      if (semver.gte(process.version, '10.0.0') || !options.http2) {
         this.listeningApp = https.createServer(options.https, app);
       } else {
-        if (!options.https.spdy) {
-          options.https.spdy = {
-            protocols: ['h2', 'http/1.1'],
-          };
-        }
+        options.https.spdy = {
+          protocols: ['h2', 'http/1.1'],
+        };
```

Comment:
Unfortunately, it is breaking change, we can't remove `spdy` option in this version, spdy, how we should rewrite this part:
```
const isHttp2 = options.http2 || options.spdy
// Log deprecation `spdy` option.

if (semver.gte(process.version, '10.0.0') || !isHttp2) {
        this.listeningApp = https.createServer(options.https, app);
      } else {
        options.https.spdy = {
          protocols: ['http/1.1'],
        };

        // Allow to using https with `http1.1`
        if (isHttp2) { 
          options.https.spdy.protocols.unshift('h2');
        }

        /* eslint-disable global-require */
        // The relevant issues are:
        // https://github.com/spdy-http2/node-spdy/issues/350
        // https://github.com/webpack/webpack-dev-server/issues/1592
        this.listeningApp = require('spdy').createServer(options.https, app);
        /* eslint-enable global-require */
      }
```

Reply:
@evilebottnawi I added the `spdy` option earlier in this PR, it wouldn't be a breaking change to remove it. Currently it looks like this:
```js
if (!options.https.spdy) {
        options.https.spdy = {
          protocols: ['h2', 'http/1.1'],
        };
      }

      if (semver.gte(process.version, '10.0.0')) {
        this.listeningApp = https.createServer(options.https, app);
      } else {
        /* eslint-disable global-require */
        // The relevant issues are:
        // https://github.com/spdy-http2/node-spdy/issues/350
        // https://github.com/webpack/webpack-dev-server/issues/1592
        this.listeningApp = require('spdy').createServer(options.https, app);
        /* eslint-enable global-require */
      }
    } else {
      this.listeningApp = http.createServer(app);
    }
```



Actually to avoid breaking changes, I think it should be like this:
```js
const isHttp2 = options.http2 || options.https.spdy;
if (options.https.spdy) {
// log deprecation warning, this means they provided it as part of https option
}

if (semver.gte(process.version, '10.0.0') || !isHttp2) {
        this.listeningApp = https.createServer(options.https, app);
      } else {

        if (!options.https.spdy) {
          options.https.spdy = {
            protocols: ['h2', 'http/1.1'],
          };
        }

        /* eslint-disable global-require */
        // The relevant issues are:
        // https://github.com/spdy-http2/node-spdy/issues/350
        // https://github.com/webpack/webpack-dev-server/issues/1592
        this.listeningApp = require('spdy').createServer(options.https, app);
        /* eslint-enable global-require */
      }
```

So basically, user should still be allowed to use this: `options.https.spdy`. It was never in the docs, but it was possible to use it before. The only breaking change now would be that they used to get http2 automatically if they were below Node 10.0.0. Now, they have to set `http2` to true to get http2.

Reply:
:+1: